plugins/passwordagentauthplugin/passwordagentplugin.cpp

```diff
@@ -535,6 +544,49 @@ Result PasswordAgentPlugin::beginAuthentication(
     return Result(Result::Pending);
 }
 
+void PasswordAgentPlugin::startDeviceLockAuthentication(
+        uint callerPid,
+        qint64 requestId,
+        const Sailfish::Secrets::InteractionParameters::PromptText &promptText)
+{
+    QTimer *timeout = new QTimer(this);
+    timeout->setInterval(120000); // 2 minutes
+    timeout->setSingleShot(true);
+    connect(timeout, &QTimer::timeout, timeout, [this, timeout, callerPid, requestId] {
+        authenticationCompleted(callerPid, requestId,
+                                Result(Result::IncorrectAuthenticationCodeError,
```

Comment:
Is `IncorrectAuthenticationCodeError` the proper error code here? Sounds like this is basically a timeout error, and the error code and message should reflect that.

Reply:
No such result code currently exists, but you're right - I will add one.